app/src/main/java/org/alphatilesapps/alphatiles/Validator.java

```diff
@@ -0,0 +1,992 @@
+                if (!validTypes.contains(gameTileTypes.get(i))) {
+                    warnings.add("row " + i + 1 + " of gametiles does not specify a valid type in the types column. Valid" +
+                            "types are " + validTypes);
+                }
+            }
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+        try {
+            langPackGoogleSheet.getTabFromName("wordlist").checkColForDuplicates(0);
+            langPackGoogleSheet.getTabFromName("wordlist").checkColForDuplicates(1);
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+        try {
+            langPackGoogleSheet.getTabFromName("settings").checkColForDuplicates(0);
+            langPackGoogleSheet.getTabFromName("settings").checkColForDuplicates(1);
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+    }
+
+    private void validateResourceSubfolders(){
+        //todo add secondary image functionality
+        for (Map.Entry<String, String> nameToMimeType : this.DESIRED_FILETYPE_FROM_SUBFOLDERS.entrySet()){
+            try {
+                DriveFolder subFolder = langPackDriveFolder.getFolderFromName(nameToMimeType.getKey());
+                subFolder.filterByMimeType(nameToMimeType.getValue());
+            } catch (ValidatorException e) {
+                fatalErrors.add(e.getMessage());
+            }
+        }
+        // in the validateResourceSubfolders() methods these booleans are set to true if it is determined
+        // that the app is trying to use these features
+        boolean hasInstructionAudio = decideIfAudioAttempted("games", 4, "(OPTIONAL)_instruction_audio");
+        if (!hasInstructionAudio){
+            DESIRED_FILETYPE_FROM_SUBFOLDERS.remove("(OPTIONAL)_instruction_audio");
+        }
+        boolean hasSyllableAudio = decideIfAudioAttempted("syllables", 4, "(OPTIONAL)_syllable_audio");
+        if (!hasSyllableAudio){
+            DESIRED_FILETYPE_FROM_SUBFOLDERS.remove("(OPTIONAL)_syllable_audio");
+        }
+        boolean hasTileAudio = decideIfAudioAttempted("gametiles", 5, "(OPTIONAL)_tile_audio");
+        if (!hasTileAudio){
+            DESIRED_FILETYPE_FROM_SUBFOLDERS.remove("(OPTIONAL)_tile_audio");
+        }
+
+        try {
+            DriveFolder resourceImages = langPackDriveFolder.getFolderFromName("resource_images");
+            ArrayList<String> resourceImageNames = langPackGoogleSheet.getTabFromName("resources").getCol(1);
+            resourceImages.checkFileNameAgainstList(resourceImageNames);
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+
+        try {
+            DriveFolder wordImages = langPackDriveFolder.getFolderFromName("word_images");
+            ArrayList<String> wordImageNames = langPackGoogleSheet.getTabFromName("wordlist").getCol(0);
```

Comment:
```suggestion
            ArrayList<String> wordsInLWC = langPackGoogleSheet.getTabFromName("wordlist").getCol(0);
```
To be clear and match with what the next try/catch block is doing